src/main/java/frc/robot/OI.java

```diff
@@ -0,0 +1,235 @@
+      Y(4),
+      LB(5), // Left Bumper
+      RB(6), // Right Bumper
+      Back(7),
+      Start(8),
+      LJ(9), // Left Joystick Button
+      RJ(10), // Right Joystick Button
+      POVUP(0),
+      POVDOWN(180),
+      POVLEFT(270),
+      POVRIGHT(90);
+
+      private final int buttonID;
+      private String buttonAction;
+
+      Button(int ID) {
+        this.buttonID = ID;
+        this.buttonAction = "";
+      }
+
+      private int getButtonID() {
+        return this.buttonID;
+      }
+      ;
+
+      private String getButtonAction() {
+        return this.buttonAction;
+      }
+
+      private void setButtonAction(String name) {
+        this.buttonAction = name;
+      }
+    };
+
+    private static final Joystick joystick = new Joystick(OI.operatorJoystickPort);
+    private static final Button kExampleButton =
+        Button.LB; // Toggles intake mode between cone and cube
+
+    public static JoystickButton getExampleButton() {
+      kExampleButton.setButtonAction("Example");
+      return new JoystickButton(joystick, kExampleButton.getButtonID());
+    }
+  }
+
+  public static void putControllerButtons() {
+    ShuffleboardLayout driverButtonsLayout =
+        Shuffleboard.getTab("Controls")
+            .getLayout("Driver Buttons", BuiltInLayouts.kList)
+            .withSize(2, 5)
+            .withPosition(0, 0)
+            .withProperties(Map.of("Label position", "HIDDEN")); // hide labels for Variables;
+
+    ShuffleboardLayout operatorButtonsLayout =
+        Shuffleboard.getTab("Controls")
+            .getLayout("Operator Buttons", BuiltInLayouts.kList)
+            .withSize(2, 5)
+            .withPosition(2, 0)
+            .withProperties(Map.of("Label position", "HIDDEN")); // hide labels for Variables;
+
+    for (Driver.Button button : Driver.Button.values()) {
+      driverButtonsLayout.add(
+          String.valueOf(button.getButtonID()),
+          "Button " + button.toString() + ": " + button.getButtonAction());
+    }
+
+    for (Operator.Button button : Operator.Button.values()) {
+      operatorButtonsLayout.add(
+          String.valueOf(button.getButtonID() + Operator.Button.values().length),
+          "Button " + button.toString() + ": " + button.getButtonAction());
+    }
+  }
+
+  public static class ControlCurve {
+    private double ySaturation; // Maximum output, in percentage of possible output
+    private double yIntercept; // Minimum output, in percentage of saturation
+    private double curvature; // Curvature shift between linear and cubic
+    private double deadzone; // Range of input that will always return zero output
+
+    public ControlCurve(double ySaturation, double yIntercept, double curvature, double deadzone) {
+      this.ySaturation = ySaturation;
+      this.yIntercept = yIntercept;
+      this.curvature = curvature;
+      this.deadzone = deadzone;
+    }
+
+    public double calculate(double input) {
+      /* https://www.desmos.com/calculator/w6ovblmmqj
```

Comment:
consider changing to a continuous function like in this desmos
https://www.desmos.com/calculator/ilq3nd4rsg

Reply:
Input immediately out of deadzone should be just enough to overcome static friction in my experience.

Reply:
Thanks Tyler.  Think this function is fine for now, and we can change to continuous after testing if needed.  Resolving